.buildkite/generate_pipeline.py

```diff
@@ -0,0 +1,115 @@
+  support is limited to key value pairs. Examples of valid plugin
+  configurations:
+  DOCKER_PLUGIN_CONFIG="devices: [ "/dev/vhost-vdpa-0" ], privileged: true"
+
+NOTE: The environment variables are specified as key value pairs, where the key
+is separated by the value through a colon followed by a space (`: `), and the
+pairs are separated by a comma followed by a space (`, `).
+"""
+
+import json
+import os
+import pathlib
+
+CONTAINER_VERSION="v12"
+DOCKER_PLUGIN_VERSION="v3.8.0"
+
+ROOT_PATH=pathlib.Path(__file__).parent.resolve()
+# The following default can be overwritten through the `X86_LINUX_AGENT_TAGS` env variable.
+DEFAULT_X86_LINUX_AGENT_TAGS="platform: x86_64.metal, os: linux"
+# The following default can be overwritten through the `AARCH64_LINUX_AGENT_TAGS` env variable.
+DEFAULT_AARCH64_LINUX_AGENT_TAGS="platform: arm.metal, os: linux"
+
+
+def retrieve_test_list(config_file):
+    with open(config_file) as jsonFile:
+        test_list = json.load(jsonFile)
+        jsonFile.close()
+    return test_list
+
+
+def get_platform_tags(platform):
+    if platform == "x86_64":
+        return os.getenv('X86_LINUX_AGENT_TAGS', DEFAULT_X86_LINUX_AGENT_TAGS)
+    elif platform == "aarch64":
+        return os.getenv('AARCH64_LINUX_AGENT_TAGS', DEFAULT_AARCH64_LINUX_AGENT_TAGS)
+
+
+def print_agent_tags(platform):
+    agent_tags = get_platform_tags(platform)
+    tags = agent_tags.split(", ")
+    for tag in tags:
+        print(f"      {tag}")
+
+
+def print_docker_plugin_config(json_docker_config):
+    for key, value in json_docker_config.items():
+        # In yaml the booleans are lower case (true, false), while in Python
+        # they're upper case (True, False). The following line is a hack
+        # to convert it to lower case.
+        if type(value) == bool:
+            value = str(value).lower()
+        print(f"          {key}: {value}")
+
+    # Check if there is any user defined docker config as well.
+    docker_plugin_config_list = os.getenv('DOCKER_PLUGIN_CONFIG', "").split(", ")
+    for docker_plugin_config in docker_plugin_config_list:
+        print(f"          {docker_plugin_config}")
+
+
+def generate_test_pipeline(config_file=f"{ROOT_PATH}/tests_description.json"):
+    test_list = retrieve_test_list(config_file)
+
+    # TODO: implement this with a class that knows about the needed indent so
+    # we don't need to count how many spaces are needed....
+    print("steps:")
+
+    for test in test_list["test_list"]:
+        for platform in test["platform"]:
+            label = "{}-{}".format(test['test_name'], platform)
+            command = test['command'].replace("{target_platform}", platform)
+            conditional = test.get("conditional", "")
+            buildkite_conditional = f"    if: {conditional}\n" if conditional else ""
+            print(
+                f"  - label: \"{label}\"\n"
+                "    commands:\n"
+                f"     - {command}\n"
```

Comment:
There's only 1 space here. Shouldn't there be 2?
```suggestion
                "    commands:\n"
                f"      - {command}\n"
```